src/com/company/zadania/obiektowe/zadanie1/Zamowienie.java

```diff
@@ -16,28 +18,64 @@ public Zamowienie(int maksRozmiar) {
     private int ileDodanych = 0;
 
     public void dodajPozycje(Pozycja p) {
+        for (Pozycja value : pozycja) {
+            if (value == null) continue;
+            if (p.getNazwaTowaru().equals(value.getNazwaTowaru())) {
```

Comment:
bezpieczniej ejst 

value.getNazwaTowaru().equals(p.getNazwaTowaru()

bo juz sprawdziles ze value != null, wiec nie bedize NPE, a tu jak Ci ktos wrzuci p majace null w nazwietowaru to lezysz z wyjatkiem